app/controllers/katello/concerns/api/v2/hosts_controller_extensions.rb

```diff
@@ -20,6 +20,14 @@ def host_collections
           @host.save!
           render(:locals => { :resource => @host }, :template => 'katello/api/v2/hosts/show', :status => 200)
         end
+
+        alias_method :foreman_index, :index
+        def index
+          foreman_index
+          @hosts = @hosts.includes([
+            :content_view, :lifecycle_environment, :subscription_facet, :content_host, :applicable_errata
+          ]).includes(content_facet: :bound_repositories)
+        end
```

Comment:
Wouldn't we have to modify the facet code to not require a tab/template file? I'm not sure this would be really straightforward, unless I'm missing something.